botocore/session.py

```diff
@@ -164,6 +165,7 @@ def _register_components(self):
         self._register_endpoint_resolver()
         self._register_event_emitter()
         self._register_response_parser_factory()
+        self._register_bucket_region_cache()
```

Comment:
I know we chatted about this, but I really don't like the session have service specific concepts in it.  The session should remain general purpose.  Otherwise if we allow service specific stuff in here, it can quickly deteriorate into a hard to maintain class.

Is there a way we can do this without have S3 concepts fixed in the session/clientcreator?  I know we want to share bucket caches across clients but it seems like there's a less hard coded way to do it.